estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,88 @@
+import math
+
+from dateutil import relativedelta
+
+from odoo import _, api, exceptions, fields, models
+
+
+class EstatePropertyOffer(models.Model):
+    _name = 'estate.property.offer'
+    _description = 'Estate property Offer'
+    _order = 'price desc'
+
+    price = fields.Float(string='Price')
+    status = fields.Selection(selection=[('accepted', 'Accepted'), ('refused', 'Refused')], copy=False)
+    partner_id = fields.Many2one('res.partner', string='Buyer', required=True)
+    property_id = fields.Many2one('estate.property', string='Property', required=True)
+    validity = fields.Integer(string='Validity (days)', default=7)
+    date_deadline = fields.Date(string='Deadline', compute='_compute_deadline', inverse='_inverse_deadline')
+    property_type_id = fields.Many2one(
+        string='Property Type',
+        related='property_id.property_type_id',
+        store=True,
+    )
+
+    _sql_constraints = [
+        ('check_price', 'CHECK(price > 0)', 'The price must be strictly positive.'),
+    ]
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        properties = {
+            p.id: p
+            for p in self.env['estate.property'].browse([
+                e.get('property_id') for e in vals_list if e.get('property_id')
```

Comment:
No need to use get a second time because it was checked first
```suggestion
                e['property_id'] for e in vals_list if e.get('property_id')
```